cpp/src/arrow/dataset/file_csv.cc

```diff
@@ -148,9 +149,14 @@ static inline Result<csv::ReadOptions> GetReadOptions(
 static inline Future<std::shared_ptr<csv::StreamingReader>> OpenReaderAsync(
     const FileSource& source, const CsvFileFormat& format,
     const std::shared_ptr<ScanOptions>& scan_options, Executor* cpu_executor) {
+#ifdef ARROW_WITH_OPENTELEMETRY
+  auto tracer = arrow::internal::tracing::GetTracer();
+  auto span = tracer->StartSpan("arrow::dataset::CsvFileFormat::OpenReaderAsync");
+#endif
```

Comment:
Could we push this `ifdef` into `StartSpan` by returning a dummy span object with no-op methods?

Reply:
We could. I didn't want to wrap too much of the API, also, I figured this would be best if people were very concerned about overhead.